src/pal/src/misc/corefx.cpp

```diff
@@ -346,3 +348,94 @@ ForkAndExecProcess(
     return success ? 0 : -1;
 }
 
+#if HAVE_STAT64 && !(defined(__APPLE__) && defined(_AMD64_))
+typedef struct stat64 stat_native;
+#elif HAVE_STAT
+typedef struct stat stat_native;
+#endif
```

Comment:
You have an #else error #endif below for stat/stat64... do you want one here, too?